compiler/src/dotty/tools/dotc/transform/localopt/Devalify.scala

```diff
@@ -159,6 +159,9 @@ class Devalify extends Optimisation {
         if (replacements.contains(t.symbol))
           deepReplacer.transform(replacements(t.symbol)).ensureConforms(t.tpe.widen)
         else t
+      case t: DefDef if timesUsed.getOrElse(t.symbol, 0) == 0 =>
```

Comment:
Does this remove only local definitions? Otherwise, it is not safe to remove public members and definitions

Reply:
Yes, the Simplifier only transforms local trees.